### PR TITLE
Add customizable section header

### PR DESCRIPTION
**Pull request by ChristopherBeltran**
Addresses https://github.com/dangoslen/dependabot-changelog-helper/issues/200.

### CODE REVIEW

**Review by dangoslen — CHANGES_REQUESTED** [review]
Overall this looks good! TY for the docs and tests.

I think we can do some better defaulting via the GitHub action file and that way better enforce someone attempting to set a blank section header value

File: __tests__/changelog-updater.test.ts [hunk 1]
```diff
@@ -79,7 +79,21 @@ test('adds section and an entry to the changelog when version exists but section
...
+### Changed
+- Bump \`package\` from v1 to v2 ([#123](https://github.com/owner/repo/pull/123))`)
+})
+
+test('adds section and an entry to the changelog when version exists but section does not and sectionHeader not provided', async () => {
+  const readable = Readable.from([CHANGELOG_MISSING_DEPENDENCIES])
+  fs.createReadStream.mockReturnValue(readable)
+  fs.readFileSync.mockReturnValue(CHANGELOG_MISSING_DEPENDENCIES)
+
+  await updateChangelog(PACKAGE_ENTRY, 'UNRELEASED', './CHANGELOG.md', 'Bump', '')
```
[the remark this answers is not in the document]
Comment (ChristopherBeltran) [reply __tests__/changelog-updater.test.ts]: Makes sense, forgot about the `action.yml` file 😅 .  Removed this.

File: src/changelog-updater.ts [hunk 2]
```diff
@@ -103,14 +106,17 @@ function addNewEntry(
...
+  sectionHeader: string
 ): void {
   // We build the entry string "backwards" so that we can only do one write, and base it on if the correct
   // sections exist
   let changelogEntry = buildEntryLine(prefix, entry)
   const lineNumber = result.lineToUpdate
   if (!result.dependencySectionFound) {
-    changelogEntry = `### Dependencies${EOL}${changelogEntry}`
+    changelogEntry = `### ${
+      sectionHeader || 'Dependencies'
```
[the remark this answers is not in the document]
Comment (ChristopherBeltran) [reply src/changelog-updater.ts]: Yep I agree, removed the default fallback.

File: src/changelog-updater.ts [hunk 3]
```diff
@@ -199,13 +205,18 @@ async function parseChangelogForEntry(
...
+  changelogPath: fs.PathLike,
+  sectionHeader: string
 ): Promise<ParsedResult> {
   const fileStream = readline.createInterface({
     input: fs.createReadStream(changelogPath),
     terminal: false
   })
 
+  const DEPENDENCY_SECTION_REGEX = new RegExp(
+    `^### (${sectionHeader}|Dependencies|DEPENDENCIES)`
```
[the remark this answers is not in the document]
Comment (ChristopherBeltran) [reply src/changelog-updater.ts]: Much cleaner 👍 Updated.

File: action.yml [hunk 4]
```diff
@@ -23,6 +23,11 @@ inputs:
       "The prefix word (after the hyphen) of the changelog entry, for example: '- [entryPrefix] `dependency` from v1.0 to v2.0'"
     required: true
     default: 'Bump'
+  sectionHeader:
```
Comment (dangoslen) [inline action.yml:26]: 👏🏼

**Review by dangoslen — APPROVED** [review]
Looks great! I'll get this merged in and cut in a `3.6.0` release soon

**Pull request merged**